src/all.test.js

```diff
@@ -90,6 +90,28 @@ describe('GOV.UK Frontend', () => {
         }, component)
       })
     })
+    it('can be initialised scoped to certain sections of the page', async () => {
```

Comment:
This makes sense, but only tests the instantiation of a single component and feels like it could be relatively fragile if we e.g. changed the way that conditional reveals work.

Wondering if there's a better way to test this, maybe as a unit test rather than as an integration test, e.g. passing a stubbed `DomElement` as the scope and asserting that `querySelectorAll` is called, and/or stubbing `document` and asserting that it doesn't get called?

(Just worth thinking about, this isn't a blocking comment 👍)

Reply:
We are somewhat bound by the limitations of our testing setup.

I'll see if I can stub an element like you've suggested...

Reply:
Totally fair, don't spend too much time on it 👍

Reply:
Yeah I think we'd have to run the test using JSDOM rather than in a browser, and it's not clear that you can spy on an element so I'll try to make the current setup clearer but not sure it's worth the time spent...

Reply:
Let me know if you think the current approach is too risky